hydromt_imod/imod.py

```diff
@@ -64,71 +66,316 @@ def __init__(
+
+        starting_head = idomain * top
+        starting_head = starting_head.where(idomain == 1)
+
+        self.set_grid(starting_head, "starting_head")
+
+    def setup_well(self):
+        # - PCR-GLOBWB netcdf (grid)
+        # - User defined, in some csv format
         pass
 
+    def setup_storage(self, specific_yield: float, specific_storage: float):
+        """
+
+        Setup confined storage values with specific storages. To account for
+        specific yield, specific yield is set in the top layer and divided by
+        the top layer's thickness.
+
+        See also: https://deltares.gitlab.io/imod/imod-python/faq/modeling.html
+
+        """
+
+        dz_top = self.grid["thickness"].sel(layer=1)
+        specific_storage_top_layer = specific_yield / dz_top
+
+        specific_storage_grid = xr.where(
+            self.grid.coords["layer"] == 1, specific_storage_top_layer, specific_storage
+        )
+
+        specific_storage_grid = specific_storage_grid.where(self.grid["idomain"] == 1)
+
+        self.set_grid(specific_storage_grid, "specific_storage")
+
+    def setup_hydraulic_conductivity(
+        self, hydraulic_conductivity: float, vertical_anisotropy=1.0
+    ):
+        # Setup K-field:
+        # - GLIMPSE?
+        # - Scalar value
+
+        kh = xr.where(self.grid["idomain"], hydraulic_conductivity, np.nan)
+        k33 = kh / vertical_anisotropy
+
+        self.set_grid(kh, "k")
+        self.set_grid(k33, "k33")
+
+    def setup_overland_flow(self, resistance=1.0):
+        """
+        Parameters
+        ----------
+        resistance: float
+            Resistance for overland flow.in days.
+            Set to 1 day, concurrent with iMOD 5.
+        """
+        # Set drain to DEM level, regrid with minimum/low percentile
+
+        elevation = self.grid["dem"].assign_coords(layer=1)
+        x = self.grid.coords["x"]
+        res = x[1] - x[0]
+        conductance_value = res**2 / resistance
+        conductance = xr.full_like(elevation, conductance_value)
+        conductance = conductance.where(~np.isnan(elevation))
+        ds = xr.merge([{"elevation": elevation, "conductance": conductance}])
+        self.set_forcing(ds, name="overland_flow", split_dataset=False)
+
+    def setup_recharge(
+        self, rate_fn: Union[str, Path] = None, rate_constant: float = None
+    ):
+        # Three options possible, from complex to least complex:
+        # - Recharge grid from wflow model (use full potential of hydromt)
+        # - P - ET
+        # - Scalar value
+
+        rate = self._raster_or_constant(
+            rate_constant, rate_fn, self.grid["grid_2d"], method="nearest"
+        )
+
+        rate = rate.assign_coords(layer=1)
+        rate = rate.where(self.grid["idomain"].sel(layer=1) == 1)
+        recharge = xr.Dataset(data_vars={"rate": rate})
+        self.set_forcing(recharge, "recharge", split_dataset=False)
+
+    def setup(self, region: dict, res: float, crs: int = "utm"):
```

Comment:
See the comments linked to test_create_model_api.py. This method is not needed as we use ``build`` and ``update`` instead, so I would delete